diff --git a/include/spdk/thread.h b/include/spdk/thread.h
index 16114b887..6d2c04474 100644
--- a/include/spdk/thread.h
+++ b/include/spdk/thread.h
@@ -1234,6 +1234,18 @@ typedef void (*spdk_iobuf_get_stats_cb)(struct spdk_iobuf_module_stats *modules,
  */
 int spdk_iobuf_get_stats(spdk_iobuf_get_stats_cb cb_fn, void *cb_arg);
 
+/* Moved from iobuf.c */
+struct iobuf {
+	struct spdk_ring		*small_pool;
+	struct spdk_ring		*large_pool;
+	void				*small_pool_base;
+	void				*large_pool_base;
+	struct spdk_iobuf_opts		opts;
+	TAILQ_HEAD(, iobuf_module)	modules;
+	spdk_iobuf_finish_cb		finish_cb;
+	void				*finish_arg;
+};
+
 #ifdef __cplusplus
 }
 #endif
diff --git a/lib/thread/iobuf.c b/lib/thread/iobuf.c
index f0a98855e..e0b177ac0 100644
--- a/lib/thread/iobuf.c
+++ b/lib/thread/iobuf.c
@@ -40,18 +40,7 @@ struct iobuf_module {
 	TAILQ_ENTRY(iobuf_module)	tailq;
 };
 
-struct iobuf {
-	struct spdk_ring		*small_pool;
-	struct spdk_ring		*large_pool;
-	void				*small_pool_base;
-	void				*large_pool_base;
-	struct spdk_iobuf_opts		opts;
-	TAILQ_HEAD(, iobuf_module)	modules;
-	spdk_iobuf_finish_cb		finish_cb;
-	void				*finish_arg;
-};
-
-static struct iobuf g_iobuf = {
+struct iobuf g_iobuf = {
 	.modules = TAILQ_HEAD_INITIALIZER(g_iobuf.modules),
 	.small_pool = NULL,
 	.large_pool = NULL,
diff --git a/mk/spdk.modules.mk b/mk/spdk.modules.mk
index 85b93436e..33afeb677 100644
--- a/mk/spdk.modules.mk
+++ b/mk/spdk.modules.mk
@@ -4,7 +4,7 @@
 #  All rights reserved.
 #
 
-BLOCKDEV_MODULES_LIST = bdev_malloc bdev_null bdev_nvme bdev_passthru bdev_lvol
+BLOCKDEV_MODULES_LIST = bdev_malloc bdev_null bdev_nvme bdev_passthru bdev_lvol bdev_integrity
 BLOCKDEV_MODULES_LIST += bdev_raid bdev_error bdev_gpt bdev_split bdev_delay
 BLOCKDEV_MODULES_LIST += bdev_zone_block
 BLOCKDEV_MODULES_LIST += blobfs blobfs_bdev blob_bdev blob lvol vmd nvme
@@ -25,6 +25,13 @@ BLOCKDEV_MODULES_PRIVATE_LIBS +=  $(XNVME_LIB_DIR)/libxnvme.a -luring -laio -pth
 endif
 endif
 
+# DOCA
+DEBUG ?= 0
+BLOCKDEV_MODULES_PRIVATE_LIBS += -L/opt/mellanox/doca/lib/aarch64-linux-gnu -Wl,--as-needed -ldoca_common -ldoca_aes_gcm -L/home/ubuntu/spdk/module/bdev/integrity -Wl,-rpath=/home/ubuntu/spdk/module/bdev/integrity -lblake3 
+ifeq ($(DEBUG),1)
+BLOCKDEV_MODULES_PRIVATE_LIBS += -fsanitize=thread -fno-omit-frame-pointer -g
+endif
+
 ifeq ($(CONFIG_VFIO_USER),y)
 BLOCKDEV_MODULES_LIST += vfio_user
 endif
diff --git a/module/bdev/Makefile b/module/bdev/Makefile
index 27aa4fa43..7d6a19646 100644
--- a/module/bdev/Makefile
+++ b/module/bdev/Makefile
@@ -6,7 +6,7 @@
 SPDK_ROOT_DIR := $(abspath $(CURDIR)/../..)
 include $(SPDK_ROOT_DIR)/mk/spdk.common.mk
 
-DIRS-y += delay error gpt lvol malloc null nvme passthru raid split zone_block
+DIRS-y += delay error gpt lvol malloc null nvme passthru raid split zone_block integrity
 
 DIRS-$(CONFIG_XNVME) += xnvme
 
